lib/controller.js

```diff
@@ -461,7 +461,7 @@ controller.renameTessel = function(opts) {
       if (!opts.reset && !opts.newName) {
         reject('A new name must be provided.');
       } else {
-        if (!Tessel.isValidName(opts.newName)) {
+        if (!opts.reset && !Tessel.isValidName(opts.newName)) {
```

Comment:
This was actually an unknown bug in `t2 rename` where the `--reset` flag would not work properly (because this if else statement rejected the command before it got to the point where it could reset)